packages/stargate/src/queries/ibc.spec.ts

```diff
@@ -0,0 +1,157 @@
+
+async function makeClientWithIbc(rpcUrl: string): Promise<[QueryClient & IbcExtension, TendermintClient]> {
+  const tmClient = await TendermintClient.connect(rpcUrl);
+  return [QueryClient.withExtensions(tmClient, setupIbcExtension), tmClient];
+}
+
+describe("IbcExtension", () => {
+  describe("unverified", () => {
+    describe("channel", () => {
+      it("can be called", async () => {
+        pending("Fails with 'Query failed with (1): internal'. Make it work.");
+        pendingWithoutSimapp();
+        const [client, tmClient] = await makeClientWithIbc(simapp.tendermintUrl);
+
+        const response = await client.ibc.unverified.channel("foo", "bar");
+        expect(response).toBeTruthy(); // TODO: implement checks
+
+        tmClient.disconnect();
+      });
+    });
+
+    describe("channels", () => {
+      it("can be called", async () => {
+        pendingWithoutSimapp();
+        const [client, tmClient] = await makeClientWithIbc(simapp.tendermintUrl);
+
+        const response = await client.ibc.unverified.channels();
+        expect(response).toBeTruthy(); // TODO: implement checks
+
+        tmClient.disconnect();
+      });
+    });
+
+    describe("connectionChannels", () => {
+      it("can be called", async () => {
+        pending("Fails with 'Query failed with (1): internal'. Make it work.");
+        pendingWithoutSimapp();
+        const [client, tmClient] = await makeClientWithIbc(simapp.tendermintUrl);
+
+        const response = await client.ibc.unverified.connectionChannels();
+        expect(response).toBeTruthy(); // TODO: implement checks
+
+        tmClient.disconnect();
+      });
+    });
+
+    describe("packetCommitment", () => {
+      it("can be called", async () => {
+        pending("Fails with 'Query failed with (1): internal'. Make it work.");
+        pendingWithoutSimapp();
+        const [client, tmClient] = await makeClientWithIbc(simapp.tendermintUrl);
+
+        const response = await client.ibc.unverified.packetCommitment();
+        expect(response).toBeTruthy(); // TODO: implement checks
+
+        tmClient.disconnect();
+      });
+    });
+
+    describe("packetCommitments", () => {
+      it("can be called", async () => {
+        pending("Fails with 'Query failed with (1): internal'. Make it work.");
+        pendingWithoutSimapp();
+        const [client, tmClient] = await makeClientWithIbc(simapp.tendermintUrl);
+
+        const response = await client.ibc.unverified.packetCommitments();
+        expect(response).toBeTruthy(); // TODO: implement checks
+
+        tmClient.disconnect();
+      });
+    });
+
+    describe("packetAcknowledgement", () => {
+      it("can be called", async () => {
+        pending("Fails with 'Query failed with (1): internal'. Make it work.");
+        pendingWithoutSimapp();
+        const [client, tmClient] = await makeClientWithIbc(simapp.tendermintUrl);
+
+        const response = await client.ibc.unverified.packetAcknowledgement();
```

Comment:
A lot of these need arguments, and need data to properly test. I will reach out to Informal Systems on this one

Reply:
> A lot of these need arguments

I added all query parameters except the pagination for lists (which I hope work with unset for the first data)